### ISSUE TITLE
If not exists

### ISSUE THREAD

**Issue by Heuristx** [pawelsalawa/sqlitestudio#4396]
SQL Studio is a very pleasant program, clean, fast and reliable.
It was hard to find anything to wish for, but there is one: when we create a new table or constraint, it would be nice to have a ckeckbox that adds the If Exists condition, like: Create table if exists.....

Thank you for a great tool!

     Steve

**Comment by pawelsalawa** [comment]
I think you meant "if not exists", but I don't think this option is viable in the GUI application, because you create one table at the time and if it already exists, you will get immediate error message and you need to fix it at once, before proceeding. To be honest I don't see any good use case for it. Unless I'm missing it, then please correct me.

**Comment by Heuristx** [comment]
Hi, Paweł,
Thank you for answering, and so quickly!
Did I write "If Exists"? That is strange, especially because as I was writing the message to you, I was typing "IF NOT EXISTS" in my code - lots of times. I can submit some cases when it is useful.
Until now I just typed the table creation in code, without using any tool. I used to use DB Browser for SQLite to see data, but it is too clumsy to create databases. Then somebody recommended your program in the B4X forum and I tried it. It makes database creation fun, but the best part of it is, when I restructure a table, it puts the data back, which is  a luxurious feature! However, when the database is created, I copy and paste the code from the DDL tabs to my code. 
The idea is that you can:
1) have only an .exe file, that simplifies installation.No need to distribute an extra database file.2) Even if the user deletes the database, the code can be sure it is there, because every time you start it, it creates all tables and populates them with the necessary minimum data. The user never has to reinstall anything.
Another case when it is useful is when you create ad-hoc tables temporarily for some reason. Like when updating from an older database to a newer one or when you want to import data from another database(with a different structure). Of course, also the index creation benefits from the IF NOT EXISTS clause.
Besides, temporary database tables can be used for other purposes than actually storing data, although this sounds a bit strange. But suppose you take over data from some source and you have to sort it with multiple columns, some descending, some ascending. You may have your sorting functions for lists or trees in your code, but typically they are written for sorting on one "column"(property and such), it is much easier to use a table, sort the data and drop the table. When it is a multithreaded program, you never know if a table was already created or not.
The same applies when you use a table and indeces to temporarily cache data for some operation.
In other words, not every program follows the pattern that you create a database, install it for the user, and never touch the structure after that.
And I learned that if I have a wish or a problem, I am never alone in the world. I can't be the only one who uses a tool in a certain way, no matter how strange that way is.
Anyway, no matter what you decide, now I have a nice program even to create data structures, view data, and test complicated queries. And I thank you for that!
All the best,
           Steve








    On Monday, February 7, 2022, 06:08:28 AM EST, Paweł Salawa ***@***.***> wrote:  
 
 


I think you meant "if not exists", but I don't think this option is viable in the GUI application, because you create one table at the time and if it already exists, you will get immediate error message and you need to fix it at once, before proceeding. To be honest I don't see any good use case for it. Unless I'm missing it, then please correct me.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by pawelsalawa** [comment]
I see. Okay, it makes sense. I'll add "if not exists" option.

Btw. you said
> then somebody recommended your program in the B4X forum and I tried it. It makes database creation fun, but the best part of it is, when I restructure a table, it puts the data back, which is  a luxurious feature!

SQLiteStudio also does put the data back to altered tables ;)

**Comment by Heuristx** [comment]
Hi, Paweł,
That is what I said, yours does! The other tools don't. And yours even shows a dialogue box with the code that does it! I wasn't used to this kind of comfort! Now I feel spoiled.
Thank you!
            Steve

    On Monday, February 7, 2022, 07:34:57 AM EST, Paweł Salawa ***@***.***> wrote:  
 
 


I see. Okay, it makes sense. I'll add "if not exists" option.

Btw. you said


then somebody recommended your program in the B4X forum and I tried it. It makes database creation fun, but the best part of it is, when I restructure a table, it puts the data back, which is a luxurious feature!


SQLiteStudio also does put the data back to altered tables ;)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by pawelsalawa** [comment]
Oh, I misread your previous post. Sorry. Somehow I read in my head "someone recommended B4X program", while the B4X is  forum.

Okay, cool. Thanks!

**Comment by Heuristx** [comment]
Right. I guess every B4X programmer is using your SQLiteStudio now. I don't know of any other program that comes close. It is fast, reliable and even looks great.It literally makes database work fun, which I would never have believed possible.

    On Monday, February 7, 2022, 07:54:57 AM EST, Paweł Salawa ***@***.***> wrote:  
 
 


Oh, I misread your previous post. Sorry. Somehow I read in my head "someone recommended B4X program", while the B4X is forum.

Okay, cool. Thanks!

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by pawelsalawa** [comment]
Thank you for kind words. I appreciate it!

**Comment by NilsHoyer** [comment]
I would like to have the "IF NOT EXISTS" feature too. 🥰
